ndb_api/db_helpers.go

```diff
@@ -165,6 +168,26 @@ func validateReqData(ctx context.Context, databaseInstanceType string, reqData m
 	return
 }
 
+// Converts a map to an action arguments list
+func convertMapToActionArguments(myMap map[string]string) []ActionArgument {
+	actionArgs := []ActionArgument{}
+	for name, value := range myMap {
+		actionArgs = append(actionArgs, ActionArgument{Name: name, Value: value})
+	}
+	return actionArgs
+}
+
+// Overwrites and appends actionArguments from database.additionalArguments to actionArguments
+func setConfiguredActionArguments(database DatabaseInterface, actionArguments map[string]string) {
+	if allowedAdditionalArguments, err := util.GetAllowedAdditionalArgumentsForType(database.GetDBInstanceType()); actionArguments != nil && err == nil {
+		for name, value := range database.GetDBInstanceAdditionalArguments() {
+			if isActionArgument, _ := allowedAdditionalArguments[name]; isActionArgument {
+				actionArguments[name] = value
+			}
+		}
+	}
```

Comment:
Think about what happens when the err!= nil. We're missing/ignoring the error here